### PR TITLE
Cato 3975

### PR DESCRIPTION
**Pull request by mario-paniccia**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/box/Validators.scala [hunk 1]
```diff
@@ -19,69 +19,69 @@ package uk.gov.hmrc.ct.box
...
 
-trait Validators[B <: BoxRetriever] {
+trait Validators {
 
   protected val boxId = getClass.getSimpleName
 
-  protected def And(predicates: ((B) => Boolean)*)(boxRetriever: B): Boolean = {
-    !predicates.exists { p => !p(boxRetriever)}
+  protected def And(predicates: (() => Boolean)*)(): Boolean = {
+     !predicates.exists { p => !p()}
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/box/Validators.scala]:
```
def existsFalsePredicate = predicates.exists(predicate => !predicate())

!existsFalsePredicate
```

File: src/main/scala/uk/gov/hmrc/ct/box/Validators.scala [hunk 2]
```diff
@@ -19,69 +19,69 @@ package uk.gov.hmrc.ct.box
...
   }
 
-  protected def Or(predicates: ((B) => Boolean)*)(boxRetriever: B): Boolean = {
-    predicates.exists { p => p(boxRetriever)}
+  protected def Or(predicates: (() => Boolean)*)(): Boolean = {
+     predicates.exists { p => p()}
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/box/Validators.scala]:
```
def existsTruePredicate = predicates.exists(predicate => predicate())

existsTruePredicate
```

File: src/main/scala/uk/gov/hmrc/ct/computations/CP285.scala [hunk 3]
```diff
@@ -30,12 +30,10 @@ case class CP285(value: Option[LocalDate]) extends CtBoxIdentifier(name = "End d
...
     val DateFormat = "dd/MM/yyyy"
 
-    collectErrors(Set(
-      requiredIf() { boxRetriever: ComputationsBoxRetriever =>
-        value.isEmpty && boxRetriever.cpQ18().value == Some(true)
-      } _,
-      cannotExistIf() { boxRetriever: ComputationsBoxRetriever => value.nonEmpty && !boxRetriever.cpQ18().orFalse },
-      { boxRetriever: ComputationsBoxRetriever =>
+    collectErrors(
+      requiredIf() { value.isEmpty && boxRetriever.cpQ18().value == Some(true)} ,
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/computations/CP285.scala]: Need a matching space on the last bracket 👍

File: src/main/scala/uk/gov/hmrc/ct/box/Validators.scala (hunk 1 above)
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/box/Validators.scala]: actually I think this is better:
protected def And(predicates: (() => Boolean)*)(): Boolean = {
     predicates.forall {_()}
  }

  protected def Or(predicates: (() => Boolean)*)(): Boolean = {
     !predicates.forall {!_()}
  }

File: src/main/scala/uk/gov/hmrc/ct/computations/CP285.scala (hunk 3 above)
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/computations/CP285.scala]: done

**Pull request merged**